src/main/java/core/basesyntax/strategy/impl/ReturnOperationHandler.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.QuantityIncrementer;
+import java.util.Map;
+
+public class ReturnOperationHandler
+        extends AbstractOperationHandler
+        implements QuantityIncrementer {
+    public ReturnOperationHandler() {
+        super(Operation.RETURN);
+    }
+
+    @Override
+    protected void processTransaction(Map<String, Integer> fruitQuantityMap,
```

Comment:
This method implementation should check the result balance before saving it in the Storage to ensure it is positive. If the balance is negative, throw a RuntimeException. This is required for handling the Purchase operation, and it's a good practice to implement similar validations for other operations like Return to maintain consistency and avoid potential issues with negative balances.